### PR TITLE
Lakscastro/pattern fixes

### PR DESCRIPTION
**Pull request by LaksCastro**
Resolves: #24

**Changelogs**

Server pattern response resolved
Create tests for response pattern
Fix last then catch syntaxe

### CODE REVIEW

**Comment by barbosamaatheus** [comment]
In the reference issue I asked you to streak the` jwt.verify()` method into **JWTManager** and add tests for the methods in this file. Could you still do that please?

**Comment by LaksCastro** [comment]
Right

File: __tests__/integrations/Session/AuthenticationUser.test.js [hunk 1]
```diff
@@ -23,10 +23,11 @@ describe("Authentication User", () => {
       password,
     });
 
-    expect(response.statusCode).toBe(200);
-    expect(response.body.user.name).toBe(name);
-    expect(response.body.user.email).toBe(email);
-    expect(response.body.token).toBeTruthy();
+    expect(response.statusCode).toBe(201);
```
Comment (barbosamaatheus) [inline __tests__/integrations/Session/AuthenticationUser.test.js:26]: I think the correct status for authentication is 200. At least it is what I see most used. I've never seen 201 ...

File: src/utils/responses.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
+module.exports = class Response {
+  constructor(res) {
+    // All available API entities
+    this.entities = {
+      USER: "user",
+      DIALOG: "dialog",
+      SERVICE: "service",
```
Comment (barbosamaatheus) [inline src/utils/responses.js:7]: why is the service an entity?

File: src/services/CreateDialogService.js [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
-    );
-
-  return response;
+  const response = new Response(res);
+
+  const { entities } = response;
+
+  try {
+    const dialog = await Dialog.create({
+      ...req.body,
```
Comment (barbosamaatheus) [inline src/services/CreateDialogService.js:12]: I believe that here we have the same problem as Topic 2. Do you agree?

File: __tests__/integrations/Session/AuthenticationUser.test.js (hunk 1 above)
Comment (LaksCastro) [reply __tests__/integrations/Session/AuthenticationUser.test.js:26]: estava 200, mas o código antigo estava como 201...

File: src/utils/responses.js (hunk 2 above)
Comment (LaksCastro) [reply src/utils/responses.js:7]: fixed!

File: src/services/CreateDialogService.js (hunk 3 above)
Comment (LaksCastro) [reply src/services/CreateDialogService.js:12]: really

File: __tests__/integrations/Session/AuthenticationUser.test.js (hunk 1 above)
Comment (LaksCastro) [reply __tests__/integrations/Session/AuthenticationUser.test.js:26]: Right, fixed!

File: src/services/AuthenticationUserService.js [hunk 4]
```diff
@@ -1,19 +1,41 @@
...
+      .code(response.INVALID_REQUEST)
+      .message("Invalid Password")
+      .send();
 
   user.password = undefined;
 
-  return res.status(200).json({ user, token: jwtGenerate({ id: user.id }) });
+  return response
+    .entity(entities.USER)
+    .code(response.SUCCESS_GET)
```
Comment (barbosamaatheus) [inline src/services/AuthenticationUserService.js:37]: It was strange to use SUCCESS GET in a post request? You might want to find a more generic name for SUCCESS_GET.

Using the names of HTTP statuses, would that be a good one? What do you think about this?

**Comment by barbosamaatheus** [comment]
> In the reference issue I asked you to streak the` jwt.verify()` method into **JWTManager** and add tests for the methods in this file. Could you still do that please?

This is already done, but could you add simple unit tests to these methods?

**Comment by LaksCastro** [comment]
Right

File: src/services/RegisterUserService.js [hunk 5]
```diff
@@ -1,25 +1,57 @@
...
+
+  const response = new Response(res);
+  const { entities } = response;
 
   try {
+    const { name, picture, email, password } = req.body;
+
     if (await User.findOne({ email }))
-      return res.status(400).json({ error: "User already exists" });
+      return response
```
Comment (barbosamaatheus) [inline src/services/RegisterUserService.js:17]: the best status for this case and 409, can you change that please?

Comment (LaksCastro) [reply src/services/RegisterUserService.js:17]: Right

**Pull request merged**